### PR TITLE
Chrome Ext: Use tab's UA in the Validator's requests

### PR DESCRIPTION
**Pull request by jamesshannon**
Currently the validator's XHR requests use the browser's "default" UA, which may be different from the tab's UA (if, e.g., it's in mobile emulation mode). If the server responds differently to different UA's this causes problems.

Fixes https://github.com/ampproject/amphtml/issues/11741

* Tab's content_script.js sends the UA with the Validation request
* Background sets this UA in its XHR request via a listener

### CODE REVIEW

File: validator/chromeextension/background.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ globals.tabToUrl = {};
 globals.validAmpBgcolor = "#ffd700";
 globals.validAmpIconPrefix = "valid";
 globals.validAmpTitle = chrome.i18n.getMessage("pagePassesValidationTitle");
+globals.userAgentHeader = 'X-AMPValidator-UA';
```
Comment (honeybadgerdontcare) [inline validator/chromeextension/background.js]: nit: alphabetize (after `globals.tabToUrl`).

File: validator/chromeextension/background.js [hunk 2]
```diff
@@ -257,6 +281,44 @@ function validateUrlFromTab(tab) {
   xhr.send();
 }
 
+/**
+ * Event handler which gets called for onBeforeSendHeaders
+ * (developer.chrome.com/extensions/webRequest#event-onBeforeSendHeaders) and
+ * updates the User-Agent header to the value that's been specified. Handler listers
```
Comment (honeybadgerdontcare) [inline validator/chromeextension/background.js]: Looks like this comment wasn't finished or it's some text that's left over: `Handler listers`

File: validator/chromeextension/manifest.json [hunk 3]
```diff
@@ -29,6 +29,8 @@
   "content_security_policy": "script-src 'self' https://cdn.ampproject.org; object-src 'self'",
   "homepage_url": "https://www.ampproject.org",
   "permissions": [
+    "webRequest",
+    "webRequestBlocking",
```
Comment (honeybadgerdontcare) [inline validator/chromeextension/manifest.json]: nit: alphabetize

Comment (jamesshannon) [reply validator/chromeextension/manifest.json]: I'm assuming you mean that w(ebRequest) comes after *(://*/*).

Comment (honeybadgerdontcare) [reply validator/chromeextension/manifest.json]: after `tabs`

**Comment by aghassemi** [comment]
handled by #12750

**Pull request closed without merging**